src/main/java/com/nokia/uwr/service/BoardServiceImpl.java

```diff
@@ -115,13 +115,15 @@ int findMaxTurn(List<UEScenario> ueScenarios) {
     @Override
     public void simulate(ScenarioSchema scenarioSchema) {
 
+        turnHandler.initializeCall(scenarioSchema.btsDescriptions());
         turnHandler.setScenario(scenarioSchema);
         IntStream.rangeClosed(1, getMaxTurns()).forEach(turn -> {
```

Comment:
It looks bad, nested method. I would make it private method, then it would be clearier.